azure-migrate/dependencies-at-scale/AzMig_Dependencies.ps1

```diff
@@ -0,0 +1,669 @@
+                    if ($ip.AddressFamily -eq [System.Net.Sockets.AddressFamily]::InterNetwork) {
+                        return "IPv4"
+                    }
+                    elseif ($ip.AddressFamily -eq [System.Net.Sockets.AddressFamily]::InterNetworkV6) {
+                        return "IPv6"
+                    }
+                    else {
+                        return "Invalid"
+                    }
+                }
+                catch {
+                    return "Invalid"
+                }
+            }
+        
+            $ipType = CheckIPAddress($iprange)
+            
+            if ($ipType -eq "IPv4" -or $ipType -eq "IPv6") {
+                $fil += "| mv-expand IPAddress=IPAddresses | extend Iprange = '$iprange' | extend result = " + $ipType.ToLower() + "_compare(tostring(IPAddress),tostring(Iprange)) | where result == '0' | project-away result,Iprange | summarize make_list(IPAddress) by tostring(ServerName),tostring(IPAddresses),tostring(Source),tostring(DependencyStatus),tostring(DependencyErrors),tostring(ErrorTimeStamp),tostring(DependencyStartTime),tostring(OperatingSystem),tostring(PowerStatus),tostring(Appliance),tostring(FriendlyNameOfCredentials),tostring(Tags),tostring(ARMID) | project-away list_IPAddress"
+            } 
+            else {
+                throw "The IP range is not valid"
+            }
+        }
+           elseif ($Key -eq "osType" -or $Key -eq "osName" -or $Key -eq "osArchitecture" -or $Key -eq "osVersion") {
+                 $fil+="| where "
+                $fil += "OperatingSystem.$Key == '$Val'"
+           }
+           elseif($Key -eq "ServerName" -or $Key -eq "Source" -or $Key -eq "DependencyStatus" -or $Key -eq "PowerStatus") {
+                 $fil+="| where "
+                $fil += "$Key == '$Val'"
+           }
+           else{
+               $fil+="| where "
+              $fil += "Tags.$Key == '$Val'"
+           }
+       }
+    }
+
+    if(-not (Test-Path -IsValid -Path $OutputCsvFile)) {
+        throw "Output CSV file path is not valid"
+    }
+
+    if (Test-path -Path $OutputCsvFile) {
+        throw "File $OutputCsvFile already exists. Specify a different name or path"
+    }
+	
+	if (-not ($OutputCsvFile -match ".*\.csv$")) {
+        throw "Output file specified is not CSV."    
+    }
+    
+    $ProjectId = Get-AzMigProject -ResourceGroupName $ResourceGroupName -ProjectName $ProjectName
+	
+       
+    $Properties = GetRequestProperties
+
+    if(-not $ProjectId) {throw "Project ID is invalid"};
+    #GetApplianceToSiteMap
+    $query = "resources| where type == 'microsoft.offazure/vmwaresites' or type == 'microsoft.offazure/hypervsites' or type == 'microsoft.offazure/serversites'| where resourceGroup == '$ResourceGroupName' and properties.discoverySolutionId has '$ProjectName'| project properties.applianceName, id"
+    $response = $null
+    $response = Search-AzGraph -Query $query
+    if (-not $response) {throw "appliance not found"}
```

Comment:
Edit the error message to "Server Discovery Solution missing Appliance Details. Invalid Solution."